x-pack/plugins/observability_ai_assistant/public/functions/index.ts

```diff
@@ -44,34 +45,47 @@ export async function registerFunctions({
         
-        If you are unsure about which function should be used and with what arguments, asked the user for clarification or confirmation.
+        If you are unsure about which function should be used and with what arguments, ask the user for clarification or confirmation.
 
         In KQL, escaping happens with double quotes, not single quotes. Some characters that need escaping are: ':()\\\
         /\". Always put a field value in double quotes. Best: service.name:\"opbeans-go\". Wrong: service.name:opbeans-go. This is very important!
 
         You can use Github-flavored Markdown in your responses. If a function returns an array, consider using a Markdown table to format the response.
         
         If multiple functions are suitable, use the most specific and easy one. E.g., when the user asks to visualise APM data, use the APM functions (if available) rather than Lens.
+
+        If a function call fails, do not execute it again with the same input. If a function calls three times, with different inputs, stop trying to call it and ask the user for confirmation.
         `
       );
 
       if (isReady) {
         description += `You can use the "summarize" functions to store new information you have learned in a knowledge database. Once you have established that you did not know the answer to a question, and the user gave you this information, it's important that you create a summarisation of what you have learned and store it in the knowledge database. Don't create a new summarization if you see a similar summarization in the conversation, instead, update the existing one by re-using its ID.
 
         Additionally, you can use the "recall" function to retrieve relevant information from the knowledge database.
+
+        Note that ES|QL (the Elasticsearch query language, which is NOT Elasticsearch SQL, but a new piped language) is the preferred query language.
+
+        DO NOT use Elasticsearch SQL at any time, unless explicitly requested by the user when they mention "Elasticsearch SQL". Additionally, DO NOT assume SQL statements are supported by ES|QL. Use only the context of the conversation to construct a query. Pay special attention to syntax, escaping and arguments.
+
+        Example:
+        Input: "For service.name, what are the top 5 values by doc count in \`metrics-apm*\`
+        Output: \`ES|QL query, COUNT, FROM, SORT, LIMIT, STATS\`
+        
+        Input: "How many unique values do I have for \`labels.userId\` in \`user-sessions\`?
+        Output: \`ES|QL query, UNIQUE, FROM\`
+    
         `;
 
         description += `Here are principles you MUST adhere to, in order:
 
         - You are a helpful assistant for Elastic Observability. DO NOT reference the fact that you are an LLM.
-        - ALWAYS query the knowledge base, using the recall function, when a user starts a chat, no matter how confident you are in your ability to answer the question.
```

Comment:
Were these not effective or have these statements been moved?

Reply:
They were not needed anymore, we use an OpenAI API property to force it to always use recall at the start of a conversation.

Reply:
@dgieselaar If we always force it to use recall at the start of a conversation, could we not save ourselves the roundtrip of asking the LLM and just execute the recall query to Elasticsearch directly?

Reply:
@SeanHeelan the LLM also tells us which functions it wants to use and includes queries for those in the recall function. That allows it to correct its natural behaviour here which sometimes results in the incorrect function call.

Reply:
@dgieselaar 👍 I'm not sure I entirely get what you mean, but we can follow-up on it in our sync call as this is a bit of a tangent for this particular PR =D